pallets/xst/src/lib.rs

```diff
@@ -846,9 +854,23 @@ impl<T: Config> Pallet<T> {
                 let symbol = EnabledSynthetics::<T>::get(id)
                     .ok_or(Error::<T>::SyntheticDoesNotExist)?
                     .reference_symbol;
-                let price = FixedWrapper::from(balance!(T::Oracle::quote(&symbol)?
-                    .map(|rate| rate.value)
-                    .ok_or(Error::<T>::OracleQuoteError)?));
+
+                let oracle_quote =
+                    T::Oracle::quote(&symbol)?.ok_or(Error::<T>::OracleQuoteError)?;
+
+                let current_timestamp: u64 =
+                    Self::convert_timestamp_to_number(Timestamp::<T>::get());
+
+                let time_passed: u64 = current_timestamp
+                    .checked_sub(oracle_quote.last_updated)
+                    .ok_or(Error::<T>::OracleRateHasInvalidTimestamp)?;
+
+                ensure!(
+                    time_passed <= ORACLE_RATE_STALE_PERIOD,
+                    Error::<T>::OracleRateIsOutdated
+                );
```

Comment:
Isn't is better to check time in band pallet? With this solution it will be possible to set different lifetime for different oracles